Building the libvirt-kubevirt docker image fails when changing QEMU user ID

When doing 'make docker' it fails to modify the QEMU user ID:

usermod: Failed to change ownership of the home directoryThe command '/bin/sh -c groupmod -g 1111 qemu && usermod  -u 1111 qemu' returned a non-zero code: 12

This is on a newly installed Fedora 26 host, building from fresh checkout of kubevirt, running simply:

```bash
# cat > hack/config-local.sh <<EOF
master_ip=192.168.122.101
primary_nic=ens3
primary_node_name=f26x86_64
docker_tag=latest
EOF
# make manifests docker
```

The full trace from the docker failed build is:

```bash
Sending build context to Docker daemon 14.34 kB
Step 1/16 : FROM kubevirt/libvirt
Trying to pull repository registry.fedoraproject.org/kubevirt/libvirt ... 
Trying to pull repository registry.access.redhat.com/kubevirt/libvirt ... 
Trying to pull repository docker.io/kubevirt/libvirt ... 
sha256:8920eb36e5bd9e1c8c999f7fc4cba99dc50dfbdc0b6a6f9a092536511ff059d4: Pulling from docker.io/kubevirt/libvirt
4db9daa7aafd: Pull complete 
f87e0a029d0e: Pull complete 
59b58aaee23a: Pull complete 
e9c2f8ccbebe: Pull complete 
d928e22d53dc: Pull complete 
f7ddebfbe55d: Pull complete 
Digest: sha256:8920eb36e5bd9e1c8c999f7fc4cba99dc50dfbdc0b6a6f9a092536511ff059d4
Status: Downloaded newer image for docker.io/kubevirt/libvirt:latest
 ---> 272bf74e52bb
Step 2/16 : MAINTAINER "The KubeVirt Project" <kubevirt-dev@googlegroups.com>
 ---> Running in bd40bb5fd046
 ---> 735cfa98e8f4
Removing intermediate container bd40bb5fd046
Step 3/16 : ENV container docker
 ---> Running in 2594eef23a02
 ---> 256385fc67c4
Removing intermediate container 2594eef23a02
Step 4/16 : RUN dnf install -y   util-linux   libcgroup-tools   ethtool   sudo   docker && dnf -y clean all
 ---> Running in 8f08afe83015
Last metadata expiration check: 0:00:14 ago on Tue Sep 26 16:04:15 2017.
Package util-linux-2.29.1-2.fc26.x86_64 is already installed, skipping.
Dependencies resolved.
================================================================================
 Package                      Arch   Version                      Repository
                                                                           Size
================================================================================
Installing:
 docker                       x86_64 2:1.13.1-22.gitb5e3294.fc26  updates  20 M
 ethtool                      x86_64 2:4.11-1.fc26                updates 136 k
 libcgroup-tools              x86_64 0.41-11.fc26                 fedora   84 k
 sudo                         x86_64 1.8.20p2-1.fc26              fedora  749 k
Installing dependencies:
 atomic-registries            x86_64 1.18.1-5.fc26                fedora   45 k
 audit                        x86_64 2.7.7-1.fc26                 fedora  244 k
 audit-libs-python3           x86_64 2.7.7-1.fc26                 fedora   78 k
 checkpolicy                  x86_64 2.6-1.fc26                   fedora  296 k
 container-selinux            noarch 2:2.22-1.fc26                updates  33 k
 container-storage-setup      noarch 0.7.0-1.git4ca59c5.fc26      updates  35 k
 docker-common                x86_64 2:1.13.1-22.gitb5e3294.fc26  updates  77 k
 docker-rhel-push-plugin      x86_64 2:1.13.1-22.gitb5e3294.fc26  updates 1.6 M
 gnupg                        x86_64 1.4.22-1.fc26                updates 1.3 M
 libnet                       x86_64 1.1.6-12.fc26                fedora   63 k
 libselinux-python3           x86_64 2.6-6.fc26                   fedora  241 k
 libsemanage-python3          x86_64 2.6-4.fc26                   fedora  112 k
 libusb                       x86_64 1:0.1.5-8.fc26               fedora   39 k
 libyaml                      x86_64 0.1.7-2.fc26                 fedora   57 k
 oci-umount                   x86_64 2:2.0.0-2.gitf90b64c.fc26    updates  30 k
 policycoreutils-python-utils x86_64 2.6-5.fc26                   fedora  218 k
 policycoreutils-python3      x86_64 2.6-5.fc26                   fedora  1.8 M
 protobuf-c                   x86_64 1.2.1-4.fc26                 fedora   34 k
 python-IPy-python3           noarch 0.81-18.fc26                 fedora   41 k
 python-rhsm-certificates     x86_64 1.20.1-1.fc26                updates  44 k
 setools-python3              x86_64 4.1.0-5.fc26                 updates 570 k
 skopeo-containers            x86_64 0.1.23-6.git1bbd87f.fc26     updates  13 k
 xfsprogs                     x86_64 4.10.0-1.fc26                fedora  1.0 M
Installing weak dependencies:
 criu                         x86_64 3.3-2.fc26                   updates 431 k
 oci-register-machine         x86_64 0-5.11.gitcd1e331.fc26       updates 1.0 M
 oci-systemd-hook             x86_64 1:0.1.13-1.gitafe4b4a.fc26   updates  35 k

Transaction Summary
================================================================================
Install  30 Packages

Total download size: 30 M
Installed size: 101 M
Downloading Packages:
--------------------------------------------------------------------------------
Total                                           1.7 MB/s |  30 MB     00:17     
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                        1/1 
  Installing       : xfsprogs-4.10.0-1.fc26.x86_64                         1/30 
  Running scriptlet: xfsprogs-4.10.0-1.fc26.x86_64                         1/30 
  Installing       : libselinux-python3-2.6-6.fc26.x86_64                  2/30 
  Installing       : libsemanage-python3-2.6-4.fc26.x86_64                 3/30 
  Installing       : container-storage-setup-0.7.0-1.git4ca59c5.fc26.no    4/30 
  Installing       : protobuf-c-1.2.1-4.fc26.x86_64                        5/30 
  Running scriptlet: protobuf-c-1.2.1-4.fc26.x86_64                        5/30 
  Installing       : libnet-1.1.6-12.fc26.x86_64                           6/30 
  Running scriptlet: libnet-1.1.6-12.fc26.x86_64                           6/30 
  Installing       : setools-python3-4.1.0-5.fc26.x86_64                   7/30 
  Installing       : skopeo-containers-0.1.23-6.git1bbd87f.fc26.x86_64     8/30 
  Installing       : python-rhsm-certificates-1.20.1-1.fc26.x86_64         9/30 
  Installing       : oci-umount-2:2.0.0-2.gitf90b64c.fc26.x86_64          10/30 
  Installing       : libusb-1:0.1.5-8.fc26.x86_64                         11/30 
  Running scriptlet: libusb-1:0.1.5-8.fc26.x86_64                         11/30 
  Installing       : gnupg-1.4.22-1.fc26.x86_64                           12/30 
  Running scriptlet: gnupg-1.4.22-1.fc26.x86_64                           12/30 
  Installing       : audit-2.7.7-1.fc26.x86_64                            13/30 
  Running scriptlet: audit-2.7.7-1.fc26.x86_64                            13/30 
  Installing       : audit-libs-python3-2.7.7-1.fc26.x86_64               14/30 
  Installing       : python-IPy-python3-0.81-18.fc26.noarch               15/30 
  Installing       : checkpolicy-2.6-1.fc26.x86_64                        16/30 
  Installing       : policycoreutils-python3-2.6-5.fc26.x86_64            17/30 
  Installing       : policycoreutils-python-utils-2.6-5.fc26.x86_64       18/30 
  Installing       : container-selinux-2:2.22-1.fc26.noarch               19/30 
  Running scriptlet: container-selinux-2:2.22-1.fc26.noarch               19/30 
setsebool:  SELinux is disabled.
libsemanage.semanage_commit_sandbox: Error while renaming /var/lib/selinux/targeted/active to /var/lib/selinux/targeted/previous. (Invalid cross-device link).
/usr/sbin/semodule:  Failed!
  Installing       : libyaml-0.1.7-2.fc26.x86_64                          20/30 
  Running scriptlet: libyaml-0.1.7-2.fc26.x86_64                          20/30 
  Installing       : atomic-registries-1.18.1-5.fc26.x86_64               21/30 
  Installing       : docker-rhel-push-plugin-2:1.13.1-22.gitb5e3294.fc2   22/30 
  Running scriptlet: docker-rhel-push-plugin-2:1.13.1-22.gitb5e3294.fc2   22/30 
  Installing       : docker-common-2:1.13.1-22.gitb5e3294.fc26.x86_64     23/30 
  Installing       : docker-2:1.13.1-22.gitb5e3294.fc26.x86_64            24/30 
  Running scriptlet: docker-2:1.13.1-22.gitb5e3294.fc26.x86_64            24/30 
  Installing       : criu-3.3-2.fc26.x86_64                               25/30 
  Running scriptlet: criu-3.3-2.fc26.x86_64                               25/30 
  Installing       : oci-register-machine-0-5.11.gitcd1e331.fc26.x86_64   26/30 
  Installing       : oci-systemd-hook-1:0.1.13-1.gitafe4b4a.fc26.x86_64   27/30 
  Installing       : sudo-1.8.20p2-1.fc26.x86_64                          28/30 
  Running scriptlet: sudo-1.8.20p2-1.fc26.x86_64                          28/30 
  Installing       : ethtool-2:4.11-1.fc26.x86_64                         29/30 
  Installing       : libcgroup-tools-0.41-11.fc26.x86_64                  30/30 
  Running scriptlet: libcgroup-tools-0.41-11.fc26.x86_64                  30/30 
  Running scriptlet: docker-2:1.13.1-22.gitb5e3294.fc26.x86_64            30/30 
  Running scriptlet: libcgroup-tools-0.41-11.fc26.x86_64                  30/30Failed to connect to bus: No such file or directory
 
  Verifying        : libcgroup-tools-0.41-11.fc26.x86_64                   1/30 
  Verifying        : ethtool-2:4.11-1.fc26.x86_64                          2/30 
  Verifying        : sudo-1.8.20p2-1.fc26.x86_64                           3/30 
  Verifying        : docker-2:1.13.1-22.gitb5e3294.fc26.x86_64             4/30 
  Verifying        : docker-common-2:1.13.1-22.gitb5e3294.fc26.x86_64      5/30 
  Verifying        : docker-rhel-push-plugin-2:1.13.1-22.gitb5e3294.fc2    6/30 
  Verifying        : atomic-registries-1.18.1-5.fc26.x86_64                7/30 
  Verifying        : libyaml-0.1.7-2.fc26.x86_64                           8/30 
  Verifying        : container-selinux-2:2.22-1.fc26.noarch                9/30 
  Verifying        : policycoreutils-python-utils-2.6-5.fc26.x86_64       10/30 
  Verifying        : policycoreutils-python3-2.6-5.fc26.x86_64            11/30 
  Verifying        : audit-libs-python3-2.7.7-1.fc26.x86_64               12/30 
  Verifying        : libselinux-python3-2.6-6.fc26.x86_64                 13/30 
  Verifying        : checkpolicy-2.6-1.fc26.x86_64                        14/30 
  Verifying        : libsemanage-python3-2.6-4.fc26.x86_64                15/30 
  Verifying        : python-IPy-python3-0.81-18.fc26.noarch               16/30 
  Verifying        : audit-2.7.7-1.fc26.x86_64                            17/30 
  Verifying        : container-storage-setup-0.7.0-1.git4ca59c5.fc26.no   18/30 
  Verifying        : gnupg-1.4.22-1.fc26.x86_64                           19/30 
  Verifying        : libusb-1:0.1.5-8.fc26.x86_64                         20/30 
  Verifying        : oci-umount-2:2.0.0-2.gitf90b64c.fc26.x86_64          21/30 
  Verifying        : python-rhsm-certificates-1.20.1-1.fc26.x86_64        22/30 
  Verifying        : skopeo-containers-0.1.23-6.git1bbd87f.fc26.x86_64    23/30 
  Verifying        : xfsprogs-4.10.0-1.fc26.x86_64                        24/30 
  Verifying        : setools-python3-4.1.0-5.fc26.x86_64                  25/30 
  Verifying        : criu-3.3-2.fc26.x86_64                               26/30 
  Verifying        : libnet-1.1.6-12.fc26.x86_64                          27/30 
  Verifying        : protobuf-c-1.2.1-4.fc26.x86_64                       28/30 
  Verifying        : oci-systemd-hook-1:0.1.13-1.gitafe4b4a.fc26.x86_64   29/30 
  Verifying        : oci-register-machine-0-5.11.gitcd1e331.fc26.x86_64   30/30 

Installed:
  docker.x86_64 2:1.13.1-22.gitb5e3294.fc26                                     
  ethtool.x86_64 2:4.11-1.fc26                                                  
  libcgroup-tools.x86_64 0.41-11.fc26                                           
  sudo.x86_64 1.8.20p2-1.fc26                                                   
  criu.x86_64 3.3-2.fc26                                                        
  oci-register-machine.x86_64 0-5.11.gitcd1e331.fc26                            
  oci-systemd-hook.x86_64 1:0.1.13-1.gitafe4b4a.fc26                            
  atomic-registries.x86_64 1.18.1-5.fc26                                        
  audit.x86_64 2.7.7-1.fc26                                                     
  audit-libs-python3.x86_64 2.7.7-1.fc26                                        
  checkpolicy.x86_64 2.6-1.fc26                                                 
  container-selinux.noarch 2:2.22-1.fc26                                        
  container-storage-setup.noarch 0.7.0-1.git4ca59c5.fc26                        
  docker-common.x86_64 2:1.13.1-22.gitb5e3294.fc26                              
  docker-rhel-push-plugin.x86_64 2:1.13.1-22.gitb5e3294.fc26                    
  gnupg.x86_64 1.4.22-1.fc26                                                    
  libnet.x86_64 1.1.6-12.fc26                                                   
  libselinux-python3.x86_64 2.6-6.fc26                                          
  libsemanage-python3.x86_64 2.6-4.fc26                                         
  libusb.x86_64 1:0.1.5-8.fc26                                                  
  libyaml.x86_64 0.1.7-2.fc26                                                   
  oci-umount.x86_64 2:2.0.0-2.gitf90b64c.fc26                                   
  policycoreutils-python-utils.x86_64 2.6-5.fc26                                
  policycoreutils-python3.x86_64 2.6-5.fc26                                     
  protobuf-c.x86_64 1.2.1-4.fc26                                                
  python-IPy-python3.noarch 0.81-18.fc26                                        
  python-rhsm-certificates.x86_64 1.20.1-1.fc26                                 
  setools-python3.x86_64 4.1.0-5.fc26                                           
  skopeo-containers.x86_64 0.1.23-6.git1bbd87f.fc26                             
  xfsprogs.x86_64 4.10.0-1.fc26                                                 

Complete!
18 files removed
 ---> f40fb7a4c250
Removing intermediate container 8f08afe83015
Step 5/16 : COPY qemu-kube /usr/local/bin/qemu-system-x86_64
 ---> fbef753c1376
Removing intermediate container b76af84e71b8
Step 6/16 : RUN chmod a+x /usr/local/bin/qemu-system-x86_64
 ---> Running in 1237591a1c18
 ---> 7ffa64800d6d
Removing intermediate container 1237591a1c18
Step 7/16 : ENV PATH /usr/local/bin:/bin:/sbin
 ---> Running in 187cfc74b3bf
 ---> fa73be8905e9
Removing intermediate container 187cfc74b3bf
Step 8/16 : COPY kubevirt-sudo /etc/sudoers.d/kubevirt
 ---> 80a6efda78ea
Removing intermediate container b1265183161e
Step 9/16 : RUN chmod 0640 /etc/sudoers.d/kubevirt
 ---> Running in 1d983ab4a686
 ---> 5dff3b34be32
Removing intermediate container 1d983ab4a686
Step 10/16 : COPY libvirtd-limited.sh /libvirtd-limited.sh
 ---> 6976a8093209
Removing intermediate container 8694bf394fbf
Step 11/16 : RUN chmod a+x /libvirtd-limited.sh
 ---> Running in b9c1d0a50b8b
 ---> af4050cf5750
Removing intermediate container b9c1d0a50b8b
Step 12/16 : RUN rm -f /libvirtd.sh
 ---> Running in 655c915be89e
 ---> be5c2176e629
Removing intermediate container 655c915be89e
Step 13/16 : COPY libvirtd.sh /libvirtd.sh
 ---> d16506829fca
Removing intermediate container f829aa77e0a5
Step 14/16 : RUN chmod a+x /libvirtd.sh
 ---> Running in b69983bb55f2
 ---> 8452f8bb6218
Removing intermediate container b69983bb55f2
Step 15/16 : RUN groupmod -g 1111 qemu && usermod  -u 1111 qemu
 ---> Running in ba16b7d43c0b
usermod: Failed to change ownership of the home directoryThe command '/bin/sh -c groupmod -g 1111 qemu && usermod  -u 1111 qemu' returned a non-zero code: 12
make: *** [Makefile:56: docker] Error 12
```

--- a/cmd/virt-handler/Dockerfile
+++ b/cmd/virt-handler/Dockerfile
@@ -21,8 +21,8 @@ FROM fedora:26
 MAINTAINER "The KubeVirt Project" <kubevirt-dev@googlegroups.com>
 
 RUN dnf -y install libvirt-client genisoimage && \
-    groupadd --gid 1111 qemu && \
-    useradd --uid 1111 --gid 1111 qemu && \
+    groupadd --gid 107 qemu && \
+    useradd --uid 107 --gid 107 qemu && \
     dnf -y clean all
 
 COPY virt-handler /virt-handler

--- a/images/libvirt-kubevirt/Dockerfile
+++ b/images/libvirt-kubevirt/Dockerfile
@@ -44,9 +44,4 @@ RUN rm -f /libvirtd.sh
 COPY libvirtd.sh /libvirtd.sh
 RUN chmod a+x /libvirtd.sh
 
-# Make a predictable uid and gid for qemu user
-# This allows containers to set enforce permissions on files that
-# are shared between containers on the local disk.
-RUN groupmod -g 1111 qemu && usermod  -u 1111 qemu
-
 CMD ["/libvirtd.sh"]

--- a/images/libvirt-kubevirt/Dockerfile
+++ b/images/libvirt-kubevirt/Dockerfile
@@ -27,7 +27,8 @@ RUN dnf install -y \
   libcgroup-tools \
   ethtool \
   sudo \
-  docker && dnf -y clean all
+  docker && dnf -y clean all && \
+  test $(id -u qemu) = 107 # make sure that the qemu user really is 107
 
 COPY qemu-kube /usr/local/bin/qemu-system-x86_64
 RUN chmod a+x /usr/local/bin/qemu-system-x86_64

